bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,269 @@
+import static org.openhab.core.types.RefreshType.REFRESH;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ScheduledFuture;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.JettyHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Cannot start server! " + ex.getMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        var config = this.getConfigAs(CloudBridgeConfig.class);
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Username or password is not valid");
+            return;
+        }
+        var httpClient = new JettyHttpClient(httpClientFactory.getCommonHttpClient());
+        var localSalusApi = salusApi = new SalusApi(config.getUsername(), config.getPassword().toCharArray(),
+                config.getUrl(), httpClient, GsonMapper.INSTANCE);
+        logger = LoggerFactory
+                .getLogger(CloudBridgeHandler.class.getName() + "[" + config.getUsername().replace(".", "_") + "]");
+
+        var scheduledPool = ThreadPoolManager.getScheduledPool("Salus");
+        scheduledPool.schedule(() -> tryConnectToCloud(localSalusApi), 1, MICROSECONDS);
+
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
```

Comment:
I've checked this API. The closest to Caffeine is `ExpiringCacheMap`, but caffeine loads (expired) values automatically without calling methods to put.

Reply:
It might not be a drop in replacement, but you can make it work with just some small changes. It would save a not needed dependency. Let me know if you need some help with: https://www.openhab.org/javadoc/latest/org/openhab/core/cache/expiringcachemap#putIfAbsentAndGet(K,java.util.function.Supplier)

Reply:
Yeah it's exactly what I don't want to do. The caller of the method should not know how to populate the cache

Reply:
Still possible with a convenience method:
```
    //free form code
    private Type GetOrSetFromCache(string key) {
        return cache.putIfAbsentAndGet(key, () -> {
                //refresh specific property or all depending on your needs. with `cache.putValue`
                return cache.get(key);
        });
    }
```

Reply:
But then you will have 2 ways of accessing cache

1. by calling method
2. by calling field 

I don't want to add this complexity into the code

Reply:
Pesonally i don't regard this as complex code, and it saves a dependency that makes it a little better to maintain. On the opther hand @magx2 will be code owner that mainly handles the maintenance. @jlaur wdyt?

Reply:
@jlaur what is your opinion?

Reply:
ping

Reply:
@openhab/add-ons-maintainers can use some advice here.